src/Icons/src/DependencyInjection/UXIconsExtension.php

```diff
@@ -0,0 +1,81 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\UX\Icons\DependencyInjection;
+
+use Symfony\Component\Config\Definition\Builder\TreeBuilder;
+use Symfony\Component\Config\Definition\ConfigurationInterface;
+use Symfony\Component\Config\FileLocator;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+use Symfony\Component\DependencyInjection\Loader\PhpFileLoader;
+use Symfony\Component\DependencyInjection\Reference;
+use Symfony\Component\HttpKernel\DependencyInjection\ConfigurableExtension;
+
+/**
+ * @author Kevin Bond <kevinbond@gmail.com>
+ *
+ * @internal
+ */
+final class UXIconsExtension extends ConfigurableExtension implements ConfigurationInterface
+{
+    public function getConfigTreeBuilder(): TreeBuilder
+    {
+        $builder = new TreeBuilder('ux_icons');
+        $rootNode = $builder->getRootNode();
+
+        $rootNode
+            ->children()
+                ->scalarNode('icon_dir')
+                    ->info('The local directory where icons are stored.')
+                    ->defaultValue('%kernel.project_dir%/templates/icons')
```

Comment:
I'm starting to think it's not a good idea.

This make impossible to expose icons, and i'm pretty sure, if we have icon packs, people would want to use them as assets. 

This also brings a lot of assets in a directory made for ".twig" files. 

What about using assets/icons and adding an ignore rule in the assetmapper conf ?

Reply:
Yeah, I was on the fence on what to do here. Here's my thinking for the current default:
1. It'd be easy enough for someone to change this to `%kernel.project_dir%/assets/icons` if they want to expose them.
2. I think we don't want them exposed by default so it'd add a bit more complexity (adding the ignore rule).
3. The latest twig recipe excludes non-twig files from being looked at

Reply:
I do agree, as we already talked about, to not send all this into assetmapper.

The problem i see there is that it creates an weird split between assets and ... assets. 
For a standard website, that'd mean icons needed to be exposed would be in a place (the logo, partner logos, idk) would be in assets/logo (preventing to use it with the icon function) but the github icon, or the partners ones would be in templates/icons ?

In a way, that make perfect sense, but with the favicons in a place, that start to look like an annoying DX (Designer Experience) :p 

(not having a perfect solution to present) 

In the end, if people download TablerIcon twice (once in assets, once in templates) we will need to reconsider this :)) But let's see, the "experimental" period can be used for that :)

Reply:
> The problem i see there is that it creates an weird split between assets and ... assets.
 Yes, exactly! This is why I was on the fence. `assets/icons/*` feels like a much better home than `templates/icons/*`.

Reply:
The "only" problem with assets/icons/* is for the massive packs (which are probably already in this folder today in many projects).

So why not start with "assets/icons" for now, and find a solution for the packs while we develop the "pack" features ? 

I see no problem having packs in specific folders, that's really the main "header / footer" icons that i'd find strange to separate from the CSS / image files.

Reply:
I'm leaning towards `assets/icons` also. Not sure, in practice, if having 1000 icons served through AssetMapper is an actual problem or not. And, as you said, perhaps the "packs" solution could fix this anyway - e.g. maybe the packs are downloaded into the `var/` directory (as they don't need be committed).

Reply:
I've made this change (it's still configurable).